zivid_camera/src/settings_generator.cpp

```diff
@@ -5,52 +5,45 @@
 
 #include <boost/algorithm/string.hpp>
 #include <boost/algorithm/string/replace.hpp>
+#include <boost/filesystem.hpp>
 
 #include <algorithm>
 #include <cctype>
 #include <chrono>
-#include <filesystem>
 #include <fstream>
 #include <regex>
 #include <sstream>
 #include <string>
 
 namespace
 {
-template <class T>
-using NormalizedType = std::remove_const_t<std::remove_reference_t<T>>;
-
-template <class T, class ZividSettingNode>
-constexpr bool is_zivid_setting = std::is_same<NormalizedType<T>, ZividSettingNode>::value;
-
-template <class T>
-constexpr bool is_filters = is_zivid_setting<T, Zivid::Settings::Filters>;
-
-template <class T>
-constexpr bool is_color_balance =
-    is_zivid_setting<T, Zivid::Settings::BlueBalance> || is_zivid_setting<T, Zivid::Settings::RedBalance>;
-
-template <class T>
-constexpr bool is_general_setting = is_filters<T> || is_color_balance<T>;
+template <typename T>
+struct DependentFalse : std::false_type
+{
+};
 
-template <typename C, typename = void>
-struct HasRange : std::false_type
+template <typename T, typename...>
+struct IsInList : std::false_type
 {
 };
 
-template <typename C>
-struct HasRange<C, std::void_t<decltype(std::declval<C>().range())>> : std::true_type
+template <typename T, typename First, typename... Rest>
+struct IsInList<T, First, Rest...>
+  : std::integral_constant<bool, std::is_same<T, First>::value || IsInList<T, Rest...>::value>
 {
 };
 
-template <typename T>
-struct DependentFalse : std::false_type
+template <typename T, typename Tuple>
+struct IsInTuple;
+
+template <typename T, typename... Ts>
+struct IsInTuple<T, std::tuple<Ts...>> : IsInList<T, Ts...>
 {
 };
 
 void writeToFile(const std::string& file_name, const std::string& text)
 {
-  if (std::filesystem::exists(file_name))
+  if (boost::filesystem::exists(file_name))
```

Comment:
Why are you using `boost::filesystem` rather than `std::filesystem`? We require a C++17 compiler

Reply:
std::filesystem does not work on gcc-6 which otherwise should be c++17 compliant, due to this error
https://stackoverflow.com/questions/39231363/fatal-error-filesystem-no-such-file-or-directory
So I swapped to boost which we already use for other stuff to avoid this problem.

Reply:
Then [this commit ](https://github.com/zivid/zivid-ros/pull/32/commits/8c9a10fa21d91868a742847bb69931ae5d387527) wouldn't build on gcc-6, I guess?

Reply:
That is true, will fix